lnrpc/signrpc/signer.proto

```diff
@@ -164,6 +164,32 @@ message TxOut {
     bytes pk_script = 2;
 }
 
+enum SignMethod {
+    /*
+    Specifies that a SegWit v0 (p2wkh, np2wkh, p2wsh) input script should be
+    signed.
+    */
+    SIGN_METHOD_SEGWIT_V0 = 0;
+
+    /*
+    Specifies that a SegWit v1 (p2tr) input should be signed by using the
+    BIP0086 method (commit to internal key only).
+    */
+    SIGN_METHOD_SEGWIT_V1_KEY_SPEND_BIP0086 = 1;
+
+    /*
+    Specifies that a SegWit v1 (p2tr) input should be signed by using a given
+    taproot hash to commit to in addition to the internal key.
+    */
+    SIGN_METHOD_SEGWIT_V1_KEY_SPEND_ROOT_HASH = 2;
+
+    /*
+    Specifies that a SegWit v1 (p2tr) input should be spent using the script
+    path and that a specific leaf script should be signed for.
+    */
+    SIGN_METHOD_SEGWIT_V1_SCRIPT_SPEND = 3;
```

Comment:
Hmm perhaps we should thread through the "taproot" naming to make the fields more descriptive to devs? So basically dropping segwitv1 for this one and the one above to: 
* `SIGN_METHOD_TAPROOT_SCRIPT_SPEND`
* `SIGN_METHOD_TAPROOT_KEY_SPEND` (dropped the root hash suffix) 
* `SIGN_METHOD_TAPROOT_KEY_SPEND_BIP0086`

Reply:
Makes sense! I renamed both the internal constants as well as the RPC enum.